File: app/src/main/java/com/divitiae/pulsesync/data/local/seed/SeedData.kt
package com.divitiae.pulsesync.data.local.seed

import com.divitiae.pulsesync.data.domain.AiSummary
import com.divitiae.pulsesync.data.domain.Article
import com.divitiae.pulsesync.data.domain.Category
import com.divitiae.pulsesync.data.domain.ResourceKind
import com.divitiae.pulsesync.data.domain.ResourceLink
import com.divitiae.pulsesync.data.domain.SentimentType

/**
 * Offline seed content in domain form. The repository writes this into Room on
 * first launch so the app is fully usable and demoable before the REST API is
 * live. It mirrors the Figma sample stories the UI was designed against; once
 * the API is deployed, live data simply overwrites these rows.
 */
object SeedData {

    val categories: List<Category> = listOf(
        Category("bursaries", "Bursaries", "Iimali zezifundo", "Beurse", isSubscribed = true),
        Category("load-shedding", "Load Shedding", "Ukucinywa kukagesi", "Beurtkrag", isSubscribed = true),
        Category("internships", "Internships", "Izifundo zomsebenzi", "Internskappe", isSubscribed = true),
        Category("technology", "Technology", "Ubuchwepheshe", "Tegnologie", isSubscribed = false),
    )

    val articles: List<Article> = listOf(
        Article(
            id = "nsfas-2027",
            headline = "NSFAS Opens 2027 Bursary Applications for First-Year Students",
            sourceName = "News24",
            sourceUrl = "https://www.news24.com/",
            author = null,
            category = "bursaries",
            imageUrl = null,
            publishedAt = 1_786_195_200_000L,
            readTimeMinutes = 4,
            summary = AiSummary(
                detailed = listOf(
                    "NSFAS has officially opened applications for the 2027 academic year, with early " +
                        "submission strongly encouraged. Returning students must reapply through the same portal.",
                    "Applicants need a certified ID copy, proof of household income, and matric results " +
                        "uploaded before the closing date; incomplete applications will not be considered.",
                ),
                condensed = listOf(
                    "NSFAS applications for 2027 are now open, early submission encouraged.",
                    "Returning students must reapply via the same portal as new applicants.",
                    "Need certified ID, proof of income, and matric results before the deadline.",
                ),
                model = "gemini-2.5-flash",
            ),
            sentiment = SentimentType.POSITIVE,
            tags = listOf("bursary", "nsfas"),
            resources = listOf(
                ResourceLink(
                    "https://www.nsfas.org.za/content/downloads/NSFAS-Application-Form.pdf",
                    "Bursary Application Form (PDF)",
                    ResourceKind.PDF,
                ),
                ResourceLink("https://my.nsfas.org.za/", "Official NSFAS Registration Portal", ResourceKind.PORTAL),
            ),
        ),
        Article(
            id = "eskom-stage-2",
            headline = "Stage 2 Load Shedding Extended Through the Weekend, Eskom Confirms",
            sourceName = "SABC News",
            sourceUrl = "https://www.sabcnews.com/",
            author = null,
            category = "load-shedding",
            imageUrl = null,
            publishedAt = 1_786_188_000_000L,
            readTimeMinutes = 3,
            summary = AiSummary(
                detailed = listOf(
                    "Eskom announced an extension of Stage 2 cuts citing unplanned breakdowns at two coal units. " +
                        "Municipal schedules remain unchanged for now.",
                    "Residents are advised to check their zone schedule via the Eskom app as " +
                        "loadshedding.eskom.co.za reflects the update.",
                ),
                condensed = listOf(
                    "Eskom extends Stage 2 load shedding through the weekend.",
                    "Cause: unplanned breakdowns at two coal-fired units.",
                    "Check your zone schedule via the Eskom app or website.",
                ),
                model = "gemini-2.5-flash",
            ),
            sentiment = SentimentType.NEGATIVE,
            tags = listOf("eskom", "stage2"),
            resources = listOf(
                ResourceLink("https://loadshedding.eskom.co.za/", "Eskom Load Shedding Schedules", ResourceKind.WEB),
            ),
        ),
        Article(
            id = "vodacom-internships",
            headline = "Vodacom Opens 2027 Graduate Internship Programme Across Four Provinces",
            sourceName = "BusinessTech",
            sourceUrl = "https://businesstech.co.za/",
            author = null,
            category = "internships",
            imageUrl = null,
            publishedAt = 1_786_177_200_000L,
            readTimeMinutes = 3,
            summary = AiSummary(
                detailed = listOf(
                    "Vodacom is accepting applications for its 12-month graduate internship, targeting " +
                        "engineering, data and commerce graduates from Gauteng, KZN, the Western and Eastern Cape.",
                    "Successful candidates receive a stipend, a mentor and a structured rotation, with " +
                        "applications closing at the end of the month.",
                ),
                condensed = listOf(
                    "12-month paid internship for engineering, data and commerce graduates.",
                    "Open to applicants in Gauteng, KZN, Western Cape and Eastern Cape.",
                    "Applications close at the end of the month via the Vodacom careers portal.",
                ),
                model = "gemini-2.5-flash",
            ),
            sentiment = SentimentType.NEUTRAL,
            tags = listOf("internship", "graduate"),
            resources = listOf(
                ResourceLink("https://www.vodacom.co.za/vodacom/careers", "Vodacom Careers Portal", ResourceKind.PORTAL),
                ResourceLink("https://www.youtube.com/", "Programme Overview Video", ResourceKind.VIDEO),
            ),
        ),
    )

    val keywords: List<String> = listOf("internship", "load shedding", "bursary")
}

File: backend/PulseSync.Api/Data/SeedData.cs
using PulseSync.Api.Models;

namespace PulseSync.Api.Data;

public static class SeedData
{
    public static readonly List<CategoryDto> Categories = new()
    {
        new("bursaries", "Bursaries", "Iimali zezifundo", "Beurse", true),
        new("load-shedding", "Load Shedding", "Ukucinywa kukagesi", "Beurtkrag", true),
        new("internships", "Internships", "Izifundo zomsebenzi", "Internskappe", true),
        new("technology", "Technology", "Ubuchwepheshe", "Tegnologie", false)
    };

    public static readonly List<ArticleDto> Articles = new()
    {
        new(
            ArticleId: "nsfas-2027",
            Headline: "NSFAS Opens 2027 Bursary Applications for First-Year Students",
            SourceName: "News24",
            SourceUrl: "https://www.news24.com/",
            Author: null,
            Category: "bursaries",
            ImageUrl: null,
            PublishedAt: 1786195200000L,
            ReadTimeMinutes: 4,
            AiSummary: new AiSummaryDto(
                Detailed: new List<string>
                {
                    "NSFAS has officially opened applications for the 2027 academic year, with early submission strongly encouraged. Returning students must reapply through the same portal.",
                    "Applicants need a certified ID copy, proof of household income, and matric results uploaded before the closing date; incomplete applications will not be considered."
                },
                Condensed: new List<string>
                {
                    "NSFAS applications for 2027 are now open, early submission encouraged.",
                    "Returning students must reapply via the same portal as new applicants.",
                    "Need certified ID, proof of income, and matric results before the deadline."
                },
                Model: "gemini-2.5-flash",
                GeneratedAt: 1786195250000L
            ),
            Sentiment: new SentimentDto(0.8, "positive"),
            DerivedTags: new List<string> { "bursary", "nsfas" },
            ExtractedLinks: new List<ExtractedLinkDto>
            {
                new("https://www.nsfas.org.za/content/downloads/NSFAS-Application-Form.pdf", "Bursary Application Form (PDF)", "PDF"),
                new("https://my.nsfas.org.za/", "Official NSFAS Registration Portal", "PORTAL")
            }
        ),
        new(
            ArticleId: "eskom-stage-2",
            Headline: "Stage 2 Load Shedding Extended Through the Weekend, Eskom Confirms",
            SourceName: "SABC News",
            SourceUrl: "https://www.sabcnews.com/",
            Author: null,
            Category: "load-shedding",
            ImageUrl: null,
            PublishedAt: 1786188000000L,
            ReadTimeMinutes: 3,
            AiSummary: new AiSummaryDto(
                Detailed: new List<string>
                {
                    "Eskom announced an extension of Stage 2 cuts citing unplanned breakdowns at two coal units. Municipal schedules remain unchanged for now.",
                    "Residents are advised to check their zone schedule via the Eskom app as loadshedding.eskom.co.za reflects the update."
                },
                Condensed: new List<string>
                {
                    "Eskom extends Stage 2 load shedding through the weekend.",
                    "Cause: unplanned breakdowns at two coal-fired units.",
                    "Check your zone schedule via the Eskom app or website."
                },
                Model: "gemini-2.5-flash",
                GeneratedAt: 1786188050000L
            ),
            Sentiment: new SentimentDto(-0.6, "negative"),
            DerivedTags: new List<string> { "eskom", "stage2" },
            ExtractedLinks: new List<ExtractedLinkDto>
            {
                new("https://loadshedding.eskom.co.za/", "Eskom Load Shedding Schedules", "WEB")
            }
        ),
        new(
            ArticleId: "vodacom-internships",
            Headline: "Vodacom Opens 2027 Graduate Internship Programme Across Four Provinces",
            SourceName: "BusinessTech",
            SourceUrl: "https://businesstech.co.za/",
            Author: null,
            Category: "internships",
            ImageUrl: null,
            PublishedAt: 1786177200000L,
            ReadTimeMinutes: 3,
            AiSummary: new AiSummaryDto(
                Detailed: new List<string>
                {
                    "Vodacom has launched its 2027 graduate recruitment drive targeting IT, engineering, and data science graduates across South Africa.",
                    "Successful candidates will rotate through telecommunications and cloud divisions during the 18-month paid development programme."
                },
                Condensed: new List<string>
                {
                    "Vodacom launches 2027 graduate internship drive for IT and engineering.",
                    "18-month paid programme rotating through cloud and telecom divisions.",
                    "Applications close at the end of the month."
                },
                Model: "gemini-2.5-flash",
                GeneratedAt: 1786177250000L
            ),
            Sentiment: new SentimentDto(0.7, "positive"),
            DerivedTags: new List<string> { "internship", "vodacom" },
            ExtractedLinks: new List<ExtractedLinkDto>
            {
                new("https://www.vodacom.com/careers/graduates", "Apply: Vodacom Graduate Portal", "PORTAL")
            }
        ),
        new(
            ArticleId: "deepseek-local-rag",
            Headline: "Running Open-Source AI Locally: Offline Edge LLMs for Constrained Connectivity",
            SourceName: "TechCentral",
            SourceUrl: "https://techcentral.co.za/",
            Author: null,
            Category: "technology",
            ImageUrl: null,
            PublishedAt: 1786162800000L,
            ReadTimeMinutes: 5,
            AiSummary: new AiSummaryDto(
                Detailed: new List<string>
                {
                    "Recent breakthroughs in quantized open models enable sub-7B parameter LLMs to run efficiently on mobile neural processing units.",
                    "This architecture allows offline text summarization and indexing during power and connectivity outages without cloud latency."
                },
                Condensed: new List<string>
                {
                    "Quantized small LLMs now run on edge mobile hardware.",
                    "Enables local AI summarization and note indexing during outages.",
                    "Reduces reliance on constant cloud connectivity."
                },
                Model: "gemini-2.5-flash",
                GeneratedAt: 1786162850000L
            ),
            Sentiment: new SentimentDto(0.5, "positive"),
            DerivedTags: new List<string> { "ai", "technology", "edge" },
            ExtractedLinks: new List<ExtractedLinkDto>
            {
                new("https://techcentral.co.za/edge-ai-south-africa", "Full Research Paper (PDF)", "PDF")
            }
        )
    };

    public static readonly Dictionary<string, FullTextDto> FullTexts = new()
    {
        ["nsfas-2027"] = new(
            ArticleId: "nsfas-2027",
            SanitizedText: "The National Student Financial Aid Scheme (NSFAS) has officially opened student applications for the 2027 academic year. Minister of Higher Education urged all prospective learners to submit early. Documentation required includes a certified copy of South African ID card or green barcode identity document, household income verification, and recent school results. Applicants from households earning less than R350,000 per annum qualify for full tuition, accommodation, and study allowances.",
            WordCount: 88,
            SizeBytes: 620,
            RetrievedAt: 1786195200000L
        ),
        ["eskom-stage-2"] = new(
            ArticleId: "eskom-stage-2",
            SanitizedText: "Eskom confirmed today that Stage 2 load shedding will remain in force over the upcoming weekend to replenish emergency generation reserves. Unplanned unit trips at Tutuka and Medupi power stations removed 1,850 MW from the national grid. The power utility requests citizens to continue switching off non-essential appliances.",
            WordCount: 52,
            SizeBytes: 390,
            RetrievedAt: 1786188000000L
        )
    };
}
